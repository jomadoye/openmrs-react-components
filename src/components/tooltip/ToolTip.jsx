import React from 'react';
import PropTypes from 'prop-types';
import '../../../assets/css/toolTip.css';

const ToolTip = ({ toolTipHeader, toolTipBody }) => (
  <span className="tooltip-text">
    {toolTipHeader && <p>{ toolTipHeader }</p>}
    <div>
      {toolTipBody && toolTipBody.map((message, index) => (
        <span key={index}>{message.toLowerCase()}</span>
      ))}
    </div>
  </span>
);

ToolTip.propTypes = {
  toolTipBody: PropTypes.arrayOf(PropTypes.string),
  toolTipHeader: PropTypes.string
};

export default ToolTip;
