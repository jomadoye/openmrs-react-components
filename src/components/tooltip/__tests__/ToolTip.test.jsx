import React from 'react';
import renderer from 'react-test-renderer';
import ToolTip from '../ToolTip';


const props = {
  tests: [ 'sampleA', 'sampleB', 'sampleC' ],
  header: "mockToolTipHeader"
};
describe('Component: TestsToolTip', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(
      <ToolTip
        toolTipBody={props.tests}
        toolTipHeader={props.header}
      />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without toolTipHeader', () => {
    const wrapper = renderer.create(
      <ToolTip
        toolTipBody={props.tests}
      />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without toolTipBody', () => {
    const wrapper = renderer.create(
      <ToolTip
        toolTipHeader={props.header}
      />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
