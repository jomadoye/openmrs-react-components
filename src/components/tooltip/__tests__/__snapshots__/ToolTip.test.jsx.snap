// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TestsToolTip should render correctly 1`] = `
<span
  className="tooltip-text"
>
  <p>
    mockToolTipHeader
  </p>
  <div>
    <span>
      samplea
    </span>
    <span>
      sampleb
    </span>
    <span>
      samplec
    </span>
  </div>
</span>
`;

exports[`Component: TestsToolTip should render without toolTipBody 1`] = `
<span
  className="tooltip-text"
>
  <p>
    mockToolTipHeader
  </p>
  <div />
</span>
`;

exports[`Component: TestsToolTip should render without toolTipHeader 1`] = `
<span
  className="tooltip-text"
>
  <div>
    <span>
      samplea
    </span>
    <span>
      sampleb
    </span>
    <span>
      samplec
    </span>
  </div>
</span>
`;
